var/spack/repos/builtin/packages/py-spacy-legacy/package.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PySpacyLegacy(PythonPackage):
+    """Legacy registered functions for spaCy backwards compatibility"""
+
+    homepage = "https://spacy.io/"
+    pypi = "spacy-legacy/spacy-legacy-3.0.12.tar.gz"
+
+    version("3.0.12", sha256="b37d6e0c9b6e1d7ca1cf5bc7152ab64a4c4671f59c85adaf7a3fcb870357a774")
+
+    depends_on("python@3.6:", type=("build", "run"))
```

Comment:
spack does not support python@3.6 and older, so this line is not needed. You can remove it if you want to make it simpler.